builtin/logical/database/backend.go

```diff
@@ -308,6 +327,7 @@ func (b *databaseBackend) clearConnection(name string) error {
 	if ok {
 		// Ignore error here since the database client is always killed
 		db.Close()
+		b.addConnectionsCounter(db.database, -1)
```

Comment:
Technically counters should only ever increase, I think you want a gauge instead.